src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1660,8 +1631,50 @@ module Crystal
         next_char
         @column_number = 1
         @line_number += 1
-        @token.type = :STRING
-        @token.value = "\n"
+
+        if delimiter_state.kind == :heredoc
+          string_end = string_end.to_s
+          old_pos    = current_pos
+          old_column = @column_number
+
+          loop do
+            if current_char != ' '
+              break
+            else
+              next_char
+            end
+          end
```

Comment:
Note that a trailing `while` is different than in Ruby. Check also #549. Ugh, we should do something about that soon.